[diag] update diag mode to newer android standard. Fixes JB#31988

philippedeswert opened the pull request.

saukko commented on src/usb_moded-dyn-config.c:
```diff
@@ -50,6 +50,10 @@ void list_item_free(mode_list_elem *list_item)
+  free(list_item->android_extra_sysfs_path3);
+  free(list_item->android_extra_sysfs_value3);
+  free(list_item->android_extra_sysfs_path4);
+  free(list_item->android_extra_sysfs_value4);
```
If value is not set in config this will blowup as the android_extra_sysfs_value4 is not set to NULL/0, right?

spiiroin replied:
To me it looks like the pointers are not initialized even if the config is there - forgot to update read_mode_file() ?

philippedeswert replied:
Had already noticed and it is fixed ;) Thanks!

philippedeswert committed:
[diag] update diag mode to newer android standard. Fixes: JB#31988

Diag mode has become a bit more complicated, so adding support
for the new interfaces.

Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>

saukko commented on rpm/usb-moded.spec:
```diff
@@ -117,6 +117,20 @@ system bus.
 This package contains the diag config for use with the android
 gadget driver.
 
+%package diag-mode-androidv5+qcom
```
Do not use + in the package name as it might break. Also I would leave the .ini files out from this for now as it causes change of functionality to existing devices. Lets see how things start forming and then consider how and where put the .ini file changes.

philippedeswert committed:
[usb-moded] Minor release 0.85.1

Signed-off-by: Philippe De Swert <philippe.deswert@jollamobile.com>

saukko commented:
We shouldn't push any hw vendor specific things like the qcom or androidvX or so to the main components, but there are other parts in usb-moded already that are bad in the same way so this won't break anything new. So merge for now :+1:
